packages/app/src/cli/services/dev.ts

```diff
@@ -524,3 +545,136 @@ async function validateCustomPorts(webConfigs: Web[], graphiqlPort: number) {
 function setPreviousAppId(directory: string, apiKey: string) {
   setCachedAppInfo({directory, previousAppId: apiKey})
 }
+
+/**
+ * Gets the tunnel or localhost config for doing app dev
+ * @param options - Options required for the config
+ * @returns A tunnel configuration object
+ */
+export async function getTunnelMode({
```

Comment:
This file is already very long, can we move this to a new file? something like `cli/services/dev/tunnel-mode.ts`

Reply:
Yeah, fair.  I wasn't sure what the CLI's preference is here.

I've moved renderPortWarnings() and getTunnelMode() into their own files and it's much nicer.